src/App.tsx

```diff
@@ -1,24 +1,118 @@
 
-const USER_ID = 0;
+const USER_ID = 10596;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [currentTodos, setCurrentTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [inputText, setInputText] = useState('');
+  const [filterType, setFilterType] = useState<FilterTypes>(FilterTypes.ALL);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [needUpdate, setNeedUpdate] = useState(false);
+
+  const handleErrorMessage = (type: string) => {
+    setErrorMessage(`Unable to ${type} a Todo`);
+  };
+
+  const handleDeleteTodo = (todoId: string) => {
+    deleteTodo(todoId)
+      .then(() => setNeedUpdate(true))
+      .catch(() => handleErrorMessage('delete'));
+  };
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then((value) => {
+        setTodos(value);
+        setNeedUpdate(false);
+      })
+      .catch(() => handleErrorMessage('get'));
+  }, [needUpdate]);
+
+  useEffect(() => {
+    if (errorMessage) {
+      setTimeout(() => {
+        setErrorMessage('');
+      }, 3000);
+    }
+  }, [errorMessage]);
+
+  useEffect(() => {
+    switch (filterType) {
+      case 'completed': {
+        setCurrentTodos(todos.filter((todo) => todo.completed));
+        break;
+      }
+
+      case 'active': {
+        setCurrentTodos(todos.filter((todo) => !todo.completed));
+        break;
+      }
+
+      default: {
+        setCurrentTodos(todos);
+      }
+    }
+  }, [todos, filterType]);
+
+  useEffect(() => {
+    if (tempTodo) {
+      addTodo(tempTodo)
+        .then((thisTodo) => {
+          setTodos([...todos, thisTodo]);
+          setTempTodo(null);
+        })
+        .catch(() => {
+          handleErrorMessage('add');
+          setTempTodo(null);
+        });
+    }
+  }, [tempTodo]);
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">React Todo App - Load Todos</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          <button type="button" className="todoapp__toggle-all active" />
+
+          <NewTodo
+            tempTodo={tempTodo}
+            userId={USER_ID}
+            inputText={inputText}
+            onTextChange={setInputText}
+            onTodoSubmit={setTempTodo}
+            onError={setErrorMessage}
+          />
+        </header>
+
+        <TodoList
+          todos={currentTodos}
+          tempTodo={tempTodo}
+          onDeleteTodo={handleDeleteTodo}
+        />
+
+        {todos.length && (
```

Comment:
Try to refresh the page and see that there is `0` hanging as a footer. You have to either cast this `todos.length` to boolean or write ``todos.length > 0`. This way you will have a boolean expression instead of 0 that won't get shown. I know we sometimes ask you to shorten the expressions, and it's better in general, but this is one case when you have to be more cautious with it
```suggestion
        {!!todos.length && (
```